posthog/warehouse/models/datawarehouse_saved_query.py

```diff
@@ -63,6 +68,7 @@ class Status(models.TextChoices):
         null=True,
         help_text="The timestamp of this SavedQuery's last run (if any).",
     )
+    credential = models.ForeignKey(DataWarehouseCredential, on_delete=models.CASCADE, null=True, blank=True)
```

Comment:
Instead of adding `credential`'s to saved query, lets add a foreign key to `DataWarehouseTable` instead, this means:
- We only have one type of "Table", regardless of whether it's an external data sourced table, or a materialized view, they should both be synonymous as far as the system is concerned
- The URL of the table is stored in the DB and not code, making easy changes to how we store the data with backward compatibility (this has been a godsend for all the DLT pipeline changes we've made)
- It'll support different S3 formats (parquet, delta, csv, etc) - this will be helpful for when we eventually move to iceberg, and again, this will be stored in the DB and so migrations of tables don't have to be one big migration

Reply:
good point let me give it a try